azure/test.yml

```diff
@@ -37,26 +37,27 @@ jobs:
       rust: ${{ parameters.rust }}
       setup: ${{ parameters.setup }}
   # Cargo options go here
-  - script: echo '##vso[task.setvariable variable=options]$(options) --all'
+  - bash: echo '##vso[task.setvariable variable=options]$(options) --all'
   - ${{ if eq(true, parameters.features)}}:
-    - script: echo '##vso[task.setvariable variable=options]$(options) --features "${{ parameters.features }}"'
+    - bash: echo '##vso[task.setvariable variable=options]$(options) --features "${{ parameters.features }}"'
+  # Add any raw args
+  - ${{ if eq(true, parameters.raw_opts) }}:
+    - bash: echo '##vso[task.setvariable variable=options]$(options) ${{ parameters.raw_opts }}'
   # Add any additional conditional libtest options to this check
-  - ${{ if or(eq(true, parameters.single_threaded), eq(true, parameters.test_ignored))}}:
-    - script: echo '##vso[task.setvariable variable=options]$(options) --'
+  # Attempts to ensure that any raw args passed do not interfere
+  - ${{ if and( or(eq(true, parameters.single_threaded), eq(true, parameters.test_ignored)), not(endsWith($(options), '--')) ) }}:
```

Comment:
I don't think you need the whole `or()` here -- it's fine to add a trailing `--` :)